File: src/main/java/org/roylance/yaorm/services/phoenix/PhoenixConnectionSourceFactory.java
package org.roylance.yaorm.services.phoenix;

import org.roylance.yaorm.services.IConnectionSourceFactory;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

/**
 * Created by mikeroylance on 10/26/15.
 */
public class PhoenixConnectionSourceFactory
    implements IConnectionSourceFactory {

    private static final String JDBCDriverName = "org.apache.phoenix.jdbc.PhoenixDriver";
    private static final String JDBCUrl = "jdbc:phoenix:%1$s:/hbase-unsecure";

    private boolean isClosed;
    private final Connection commonConnection;

    public PhoenixConnectionSourceFactory(final String host) throws ClassNotFoundException, SQLException {
        Class.forName(JDBCDriverName);
        final String jdbcUrl = String.format(JDBCUrl, host);
        this.commonConnection = DriverManager.getConnection(jdbcUrl);
    }

    @Override
    public Connection getConnectionSource() throws SQLException {
        if (this.isClosed) {
            throw new SQLException("already closed...");
        }
        return this.commonConnection;
    }

    @Override
    public void close() throws Exception {
        if (!this.isClosed) {
            this.commonConnection.close();
        }
        this.isClosed = true;
    }
}

File: java/src/main/java/org/roylance/yaorm/services/phoenix/PhoenixConnectionSourceFactory.kt
package org.roylance.yaorm.services.phoenix

import org.roylance.yaorm.services.IConnectionSourceFactory

import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

class PhoenixConnectionSourceFactory @Throws(ClassNotFoundException::class, SQLException::class)
constructor(host: String) : IConnectionSourceFactory {

    private var isClosed: Boolean = false
    private val commonConnection: Connection

    init {
        Class.forName(JDBCDriverName)
        val jdbcUrl = String.format(JDBCUrl, host)
        this.commonConnection = DriverManager.getConnection(jdbcUrl)
    }

    override val connectionSource: Connection
        @Throws(SQLException::class)
        get() {
            if (this.isClosed) {
                throw SQLException("already closed...")
            }
            return this.commonConnection
        }

    @Throws(Exception::class)
    override fun close() {
        if (!this.isClosed) {
            this.commonConnection.close()
        }
        this.isClosed = true
    }

    companion object {

        private val JDBCDriverName = "org.apache.phoenix.jdbc.PhoenixDriver"
        private val JDBCUrl = "jdbc:phoenix:%1\$s:/hbase-unsecure"
    }
}